simulator/session_manager.go

```diff
@@ -61,16 +72,35 @@ func (s *SessionManager) Login(login *types.Login) soap.HasFault {
 				Locale:         login.Locale,
 				MessageLocale:  login.Locale,
 			},
+			Registry: NewRegistry(),
+		}
+
+		ctx.SetSession(session)
+
+		body.Res = &types.LoginResponse{
+			Returnval: session.UserSession,
 		}
 	}
 
 	return body
 }
 
-func (s *SessionManager) Logout(*types.Logout) soap.HasFault {
+func (s *SessionManager) Logout(ctx *Context, _ *types.Logout) soap.HasFault {
+	s.remove(ctx.Session.Key)
 	return &methods.LogoutBody{Res: new(types.LogoutResponse)}
 }
 
+func (s *SessionManager) TerminateSession(ctx *Context, req *types.TerminateSession) soap.HasFault {
+	for _, id := range req.SessionId {
+		if id == ctx.Session.Key {
+			return &methods.TerminateSessionBody{Fault_: Fault("", &types.InvalidArgument{})}
+		}
+		s.remove(id)
```

Comment:
do we want to check first then remove to avoid removing half of the sessions?

Reply:
Real ESX/VC actually behaves the same way as this.